arroyo-datastream/src/lib.rs

```diff
@@ -36,6 +39,38 @@ use rand::rngs::SmallRng;
 use rand::{Rng, SeedableRng};
 use regex::Regex;
 
+pub const TIMESTAMP_FIELD: &str = "_timestamp";
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ArroyoSchema {
+    pub schema: Arc<Schema>,
+    pub timestamp_col: usize,
+    pub key_cols: Vec<usize>,
```

Comment:
very minor nit: I like `key_indices` a bit better.

Reply:
👍  makes sense to reserve "col" for the actual array